src/libraries/System.Private.CoreLib/src/System/Globalization/TextInfo.cs

```diff
@@ -52,9 +52,14 @@ internal TextInfo(CultureData cultureData)
             }
         }
 
-        private TextInfo(CultureData cultureData, bool readOnly)
+        private TextInfo(CultureData cultureData, bool readOnly, bool isInvariant = false)
```

Comment:
Rather than changing the constructor and making all TextInfo constructions do the check, can the above call site instead just be:
```C#
internal static readonly TextInfo Invariant = new TextInfo(CultureData.Invariant, readOnly: true) { _isAsciiCasingSameAsInvariant = true };
```
?

Reply:
Done